public/lessons/greedy/exchange-arguments.mdx

```diff
@@ -0,0 +1,80 @@
+# Exchange Argument
+An exchange argument is a technique used to prove the optimality of greedy algorithms. It involves showing that any non-optimal solution can be transformed into an optimal one by exchanging elements in a strategic manner.
+
+### Steps to Apply Exchange Arguments
+1. **Identify Greedy Choices**: Identify the key decisions made by the greedy algorithm at each step.
+2. **Define Exchange Strategy**: Define a strategy for exchanging elements to transform any suboptimal solution into an optimal one.
+3. **Prove Optimality**: Use mathematical induction or contradiction to prove that the exchange strategy always leads to an optimal solution.
+
+## Example Problem
+Let's consider a classic example where exchange arguments can be applied: the Activity Selection Problem.
+
+### Activity Selection Problem
+Given a set of activities each marked by a start time and finish time, the goal is to select the maximum number of non-overlapping activities.
+
+### Greedy Algorithm
+1. Sort activities by finish time.
+2. Select the first activity from the sorted list of activities by earliest finish time.
+3. For each remaining activity, if its start time is after the finish time of the last selected activity, select it.
+
+### Exchange Argument
+1. **Identify Greedy Choices**: The greedy choice is to always select the activity with the earliest finish time that doesn't conflict with previously selected activities.
+2. **Define Exchange Strategy**: Suppose there exists an optimal solution where the greedy algorithm fails. We can exchange the conflicting activity in this solution with the one selected by the greedy algorithm. This exchange will not decrease the number of selected activities, and it might even increase it, contradicting the optimality of the assumed solution.
+3. **Prove Optimality**: Show that the exchange strategy preserves or improves the solution's optimality, thus proving the optimality of the greedy algorithm.
+
+### A Common Deception
+- Activity A: (1, 5)
+- Activity B: (2, 3)
+- Activity C: (4, 7)
+
+```
+A: ---------
+B:   ---
+C:       -------
+   1 2 3 4 5 6 7 8 9
+```
+
+### Deception Unveiled
+At first glance, it may seem that selecting Activity A first (since it starts earliest) is a good greedy choice. However, by doing so, we eliminate the possibility of selecting Activity B or Activity C, which would result in a suboptimal solution.
+
+## Another Example Problem
+
+### Fractional Knapsack Problem
+Given a set of items, each with a weight \( w_i \) and a value \( v_i \), and a knapsack with a weight capacity \( W \), the goal is to maximize the total value of items in the knapsack without exceeding its weight capacity. Unlike 0/1 Knapsack, fractional knapsack allows fractions of items to be taken.
```

Comment:
I don't think we should mention 0/1 knapsack yet. Most people will be looking at the greedy lesson before knowing D.P. If it is mentioned, it should be mentioned like "One variation of this problem is where you must take the full item, which cannot be solved with a greedy technique."